test/reflection.jl

```diff
@@ -813,3 +813,13 @@ f20872(::Val, ::Val) = false
 module M29962 end
 # make sure checking if a binding is deprecated does not resolve it
 @test !Base.isdeprecated(M29962, :sin) && !Base.isbindingresolved(M29962, :sin)
+
+# issue #29100
+function f29100(data)
+    names = propertynames(data)
+    map(
+        name -> getproperty(data, name),
+        names
+    )
```

Comment:
That test is impressive. I didn't expect we could get constants through map like that. Is there a limit on the tuple size for this to work?

Reply:
I didn't realize it at the time, but yes, this really only works because via the tuple is so small that map doesn't need to recur, so constant propagation isn't disabled. If I was going to revisit this, I'd probably return names wrapped in `Val` or something